syncd/scripts/syncd_init_common.sh

```diff
@@ -211,14 +211,28 @@ config_syncd_mlnx()
     DUAL_TOR="$(echo $SYNCD_VARS | jq -r '.dual_tor')"
     DSCP_REMAPPING="$(echo $SYNCD_VARS | jq -r '.dscp_remapping')"
 
-    # Make default sai.profile
+    SAI_COMMON_FILE_PATH=/usr/bin/sai-profile-common
+
+    if [[ -f $SAI_COMMON_FILE_PATH ]]; then
+        cat $SAI_COMMON_FILE_PATH > /tmp/sai.profile
+    fi
+
     if [[ -f $HWSKU_DIR/sai.profile.j2 ]]; then
         export RESOURCE_TYPE="$(echo $SYNCD_VARS | jq -r '.resource_type')"
-        j2 -e RESOURCE_TYPE $HWSKU_DIR/sai.profile.j2 -o /tmp/sai.profile
+        j2 -e RESOURCE_TYPE $HWSKU_DIR/sai.profile.j2 -o /tmp/sai-temp.profile
+
+        cat /tmp/sai-temp.profile >> /tmp/sai.profile
+        rm /tmp/sai-temp.profile
     else
-        cat $HWSKU_DIR/sai.profile > /tmp/sai.profile
+        cat $HWSKU_DIR/sai.profile >> /tmp/sai.profile
     fi
 
+    # Reverse /tmp/sai.profile lines,
+    # keep only the last (first after reverse) occurence of each prefix with '=' sign, and remove the others.
+    # finally, reverse lines again.
+
+    tac /tmp/sai.profile | awk -F= '!seen[$1]++' | tac > /tmp/sai-temp.profile && mv /tmp/sai-temp.profile /tmp/sai.profile
```

Comment:
@noaOrMlnx why is this part so complicated? Suggest having:
```
awk '!seen[$0]++' /tmp/sai-temp.profile > /tmp/sai.profile
rm -f /tmp/sai-temp.profile
```

Reply:
Done